src/components/DataViewer/index.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  createStyles,
+  Theme,
+  Grid,
+  Paper,
+  WithStyles,
+  withStyles,
+  IconButton,
+  Button,
+  CircularProgress,
+} from '@material-ui/core';
+import { Close } from '@material-ui/icons';
+import {
+  datasetSelector,
+  loadingDatasetSelector,
+  clearDataset,
+  loadDataset,
+  DatasetParams,
+  updateAdminId,
+} from '../../context/datasetStateSlice';
+import { dateRangeSelector } from '../../context/mapStateSlice/selectors';
+import Chart from '../DataDrawer/Chart';
+import { ChartConfig } from '../../config/types';
+
+function DataViewer({ classes }: DatasetProps) {
+  const dispatch = useDispatch();
+  const isDatasetLoading = useSelector(loadingDatasetSelector);
+  const { startDate: selectedDate } = useSelector(dateRangeSelector);
+
+  const {
+    data: dataset,
+    boundaryProps,
+    id,
+    serverParams,
+    title,
+    chartType,
+  } = useSelector(datasetSelector);
+
+  useEffect(() => {
+    if (boundaryProps && serverParams && id && selectedDate) {
+      const datasetParams: DatasetParams = {
+        id,
+        boundaryProps,
+        url: serverParams.url,
+        serverLayerName: serverParams.layerName,
+        selectedDate,
+      };
+
+      dispatch(loadDataset(datasetParams));
+    }
+  }, [id, boundaryProps, dispatch, serverParams, selectedDate]);
+
+  if (
+    !boundaryProps ||
+    !dataset ||
+    !serverParams ||
+    !title ||
+    !id ||
+    !chartType
+  ) {
+    return null;
+  }
+
+  const config: ChartConfig = {
+    type: chartType,
+    stacked: false,
+    fill: false,
+    category: id,
+  };
+
+  const boundaryButtons = Object.entries(boundaryProps).map(
+    ([adminId, level]) => (
+      <Button
+        id={adminId}
+        className={classes.adminButton}
+        onClick={() => dispatch(updateAdminId(adminId))}
+        size="small"
+        color="primary"
+        variant={id === adminId ? 'contained' : 'text'}
+      >
+        {level.name}
+      </Button>
+    ),
+  );
```

Comment:
In the designs, it's not clear at first glance what the names are. Can we try and differentiate by level a bit more?